tests/main/security-private-tmp/task.yaml

```diff
@@ -1,51 +1,56 @@
 summary: Ensure that the security rules for private tmp are in place.
-
-# ppc64el disabled because of https://bugs.launchpad.net/snappy/+bug/1655594
-systems: [-ubuntu-core-*, -ubuntu-*-ppc64el]
-
-environment:
-    SNAP_INSTALL_DIR: $(pwd)/snap-install-dir
-
 prepare: |
-    echo "Given a basic snap is installed"
-    #shellcheck source=tests/lib/snaps.sh
-    . "$TESTSLIB"/snaps.sh
-    install_local test-snapd-tools
-
-    echo "And another basic snap is installed"
-    mkdir -p "$SNAP_INSTALL_DIR"
-    cp -ra "$TESTSLIB"/snaps/test-snapd-tools/* "$SNAP_INSTALL_DIR"
-    sed -i 's/test-snapd-tools/not-test-snapd-tools/g' "$SNAP_INSTALL_DIR/meta/snap.yaml"
-    snap pack "$SNAP_INSTALL_DIR"
-    snap install --dangerous not-test-snapd-tools_1.0_all.snap
-
+    snap pack test-snapd-tmp-a
+    snap pack test-snapd-tmp-b
+    snap install --dangerous test-snapd-tmp-a_1_all.snap
+    snap install --dangerous test-snapd-tmp-b_1_all.snap
 restore: |
-    rm -rf "$SNAP_INSTALL_DIR" /tmp/foo
-
+    snap remove test-snapd-tmp-a
+    snap remove test-snapd-tmp-b
+    rm -f test-snapd-tmp-{a,b}-1_all.snap
 execute: |
-    #shellcheck source=tests/lib/dirs.sh
+    # shellcheck source=tests/lib/dirs.sh
     . "$TESTSLIB"/dirs.sh
 
-    echo "When a temporary file is created by one snap"
-    expect -d -f tmp-create.exp
-
-    if [ -e "$LIBEXECDIR/snapd/snap-discard-ns" ]; then
-        echo "Then that file is accessible from other calls of commands from the same snap"
-        if ! test-snapd-tools.cmd stat /tmp/foo 2>same-stat.error; then
-            echo "Expected the file to be present"
-            exit 1
-        fi
-    else
-        echo "Then that file is not accessible from other calls of commands from the same snap"
-        if test-snapd-tools.cmd stat /tmp/foo 2>same-stat.error; then
-            echo "Expected the file to be absent"
-            exit 1
-        fi
-    fi
-
-    echo "And that file is not accessible by other snaps"
-    if not-test-snapd-tools.cmd stat /tmp/foo 2>other-stat.error; then
-        echo "Expected error not present"
-        exit 1
-    fi
-    MATCH "stat: cannot stat '/tmp/foo': No such file or directory" < other-stat.error
+    # Each snap gets a private /tmp directory
```

Comment:
Maybe it's worth checking whether `/tmp/snap.test-snapd-tmp-{a,b}` don't exist before you run the respective snaps.